sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/QueryStageExec.scala

```diff
@@ -154,14 +154,14 @@ abstract class QueryStageExec extends LeafExecNode {
 abstract class ExchangeQueryStageExec extends QueryStageExec {
 
   /**
-   * Cancel the stage materialization if in progress; otherwise do nothing.
+   * Cancel the stage materialization if in progress with a reason; otherwise do nothing.
    */
-  final def cancel(): Unit = {
+  final def cancel(reason: String): Unit = {
     logDebug(s"Cancel query stage: $name")
-    doCancel()
+    doCancel(Option(reason))
   }
 
-  protected def doCancel(): Unit
+  protected def doCancel(reason: Option[String]): Unit
```

Comment:
where do we call `doCancel` without a reason?

Reply:
I don't think we ever call it without a reason. Changed the type from `Option[String]` to `String`.